test/apps/iodemo/io_demo.cc

```diff
@@ -2664,6 +2670,35 @@ static void adjust_opts(options_t *test_opts) {
         test_opts->chunk_size = std::min(test_opts->chunk_size,
                                          test_opts->max_data_size);
     }
+
+    for (size_t idx = 0; idx < servers.size(); idx++) {
+        port_separator = strchr(servers[idx], ':');
+        if (port_separator == NULL) {
+            test_opts->servers.push_back(servers[idx]);
+            continue;
+        }
+
+        repeat_separator = strchr(port_separator + 1, ':');
+        if (repeat_separator == NULL) {
+            test_opts->servers.push_back(servers[idx]);
+            continue;
+        }
+
+        server_repeated =
+            std::string(servers[idx], repeat_separator - servers[idx]);
+        repeat_count    = atoi(repeat_separator + 1);
+        for (int i = 0; i < repeat_count; i++) {
```

Comment:
To solve Coverity error, probably need to set upper bound for repeat_count, for example
```cpp
#define MAX_SERVER_REPEAT_COUNT 100000ul
...
int parse_server(const char *server_arg) {
    ...
    repeat_count = atoi(repeat_separator + 1);
    if (repeat_count > MAX_SERVER_REPEAT_COUNT) {
        std::cout << "Exceeded maximal server repeat count (" << MAX_SERVER_REPEAT_COUNT << ")" << std::endl;
        return -1;
    }
```

Reply:
Done. I also check whether the ```repeat_count``` is ```0``` here to 1) check wrong input 2) ```0``` should not be used by client.